data-prepper-plugins/aggregate-processor/src/main/java/com/amazon/dataprepper/plugins/processor/aggregate/AggregateActionResponse.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.amazon.dataprepper.plugins.processor.aggregate;
+
+import com.amazon.dataprepper.model.event.Event;
+
+import java.util.Optional;
+
+/**
+ * Model class to be returned in {@link com.amazon.dataprepper.plugins.processor.aggregate.AggregateAction}. Contains both the Event to be processed and an option
+ * to close the current window for {@link com.amazon.dataprepper.plugins.processor.aggregate.AggregateProcessor} immediately after an event is handled.
+ * @since 1.3
+ */
+public class AggregateActionResponse {
+    private Optional<Event> event;
+    private boolean closeWindowNow;
+
+    public AggregateActionResponse(Optional<Event> event, boolean closeWindowNow) {
```

Comment:
The items in this response can change overtime, so a builder pattern will help here.

But, I'd prefer that we start by providing simple functions for common use-cases:

e.g. With no event:

```
AggregateActionResponse.emptyEventResponse()
```

Or with the Event:

```
AggregateActionResponse.fromEvent(event);
```

Then as the fields grows, we can add a full-on builder pattern instead.